### ISSUE TITLE
form field -> description-field usage as description, maybe better accessibility

### ISSUE THREAD

**Issue by Teisi** [in2code-de/powermail#1240]
Can we treat the db field of a form field “description” differently?
Currently the field “description” is displayed as label-title-attribute, which I find confusing because the field is called “description” but is used as title and also because this field is not displayed visibly. This also means that no further information can be displayed, for example: what should be entered in the field or in what form.

Currently for example:

```
<label for="powermail_field_url" class="form-label powermail_label" title="My Field Description">Url</label>
<input .... />
```

I would suggest improving this by changing this to:

```
<label for="powermail_field_url" class="form-label powermail_label" aria-labelledby="powermail_field_description_{field.marker}">Url</label>
<input .... />
<div id="powermail_field_description_{field.marker}">{field.description}</div>
```

![Image](https://github.com/user-attachments/assets/cb1320fc-ecd6-4678-b2e5-a7f875b56ad1)

And if you really want to keep a title attribute, provide a separate new field for it.

"Conversely, content being put in the title attribute is being hidden from the (probable) majority of your users. If information is being hidden from the majority of your users, then it's likely not necessary."
https://www.a11yproject.com/posts/title-attributes/

**Comment by deoostfrees** [comment]
Hi @Teisi,

great suggestion. I would integrate it as follows:

```
<label for="powermail_field_url" class="form-label powermail_label">Url</label>
<input type="text" id="powermail_field_url" aria-describedby="powermail_field_description_{field.marker}" />
<p id="powermail_field_description_{field.marker}">{field.description}</p>
```

See also [Example 1: Using aria-describedby to associate instructions with form fields](https://www.w3.org/WAI/WCAG22/Techniques/aria/ARIA1).

**Comment by deoostfrees** [comment]
Hi @Teisi,

I've implemented it, and it will be included in the next major update.

**Comment by Teisi** [comment]
Thank you very much.
I took the FE Fluid Templates from the current 14.1.1 powermail version and put them into the 13.1.0 version (we currently can not use TYPO3 14...) — at first glance, it seems to be working pretty well. Thanks. :)

**Comment by nebrot** [comment]
With current solution in v14 aria-describedby is always set which is not correct when there is no description for a field. WAVE output: "Broken ARIA reference".

**Comment by deoostfrees** [comment]
@Teisi Great! I'll close the ticket here, then.
@nebrot I've created a new ticket for this issue and am looking into it.
